### PR TITLE
chore: upgrade webpack 5

### PR DESCRIPTION
**Pull request by jmfrancois**
**What is the problem this PR is trying to solve?**

* we use the outdated webpack 4
* it brings security issue:
webpack4 => watchpack => ... => glob-parent
* lack of features (web worker, module federation)

**What is the chosen solution to this problem?**

* Upgrade to webpack 5
* Upgrade all loader

QA validations tasks:
* yarn list --pattern webpack is as flat as possible ✅ 
* yarn list --pattern loader is as flat as possible ✅
* yarn postinstall ✅ (icons, tokens, theme are OK)
* yarn run pre-release ✅ (check dependencies.json are generated along report and map files)

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [x] Docs have been added / updated (for bug fixes / features)
- [x] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

**Comment by frassinier** [comment]
Still something to fix 😩 

http://4005.talend.surge.sh/design-system/?path=/docs/component-status--page&globals=theme:light

File: package.json [hunk 1]
```diff
@@ -34,7 +34,14 @@
...
   "resolutions": {
     "@types/react": "^16.14.25",
-    "@types/react-dom": "^16.9.14"
+    "@types/react-dom": "^16.9.14",
+    "@storybook/addons": "^6.5.0-beta.8",
+    "webpack": "^5.72.0",
+    "@storybook/react/webpack": "^5.72.0",
+    "@storybook/core-common/webpack": "^5.72.0",
+    "@storybook/core-server/webpack": "^5.72.0",
+    "@storybook/builder-webpack4/webpack": "^5.72.0",
```
Comment (smouillour) [inline package.json:43]: is it for webpack5 compatibility?

Comment (smouillour) [reply package.json:43]: is it still needed if we set it to webpack 5 in DS?
https://github.com/Talend/ui/blob/732924abac8e5ea66ed025d085d6f8b6c9956153/packages/design-tokens/.storybook/main.js#L8

Comment (jmfrancois) [reply package.json:43]: yes indeed. storybook do not want to create a 7.0 version so they let webpack 4 as a dependency ...
We do not have other choice than set the builder and force resolution. Which means all project will have to do so.
Quite disappointed also by their choice.

**Review by smouillour — APPROVED** [review]

**Review by frassinier — APPROVED** [review]

File: packages/icons/webpack.config.js [hunk 2]
```diff
@@ -35,10 +48,8 @@ module.exports = {
...
-	plugins: [
-		new MiniCssExtractPlugin({ filename: 'talend-icons-webfont.css' }),
-	],
-	node: {
-		fs: 'empty',
-	},
+	plugins: [new MiniCssExtractPlugin({ filename: 'talend-icons-webfont.css' })],
+	// node: {
+	// 	fs: 'empty',
+	// },
```
Comment (Kporal) [inline packages/icons/webpack.config.js:54]:
```suggestion
```

File: packages/icons/webpack.umd.js [hunk 3]
```diff
@@ -37,9 +38,9 @@ module.exports = (env = {}) => ({
 			],
 		}),
 	],
-	node: {
-		fs: 'empty',
-	},
+	// node: {
+	// 	fs: 'empty',
+	// },
```
Comment (Kporal) [inline packages/icons/webpack.umd.js:43]:
```suggestion
```

File: tools/scripts-config-react-webpack/config/webpack.config.umd.js [hunk 4]
```diff
@@ -106,9 +118,9 @@ module.exports = options => {
 					},
 				],
 			},
-			node: {
-				fs: 'empty',
-			},
+			// node: {
+			// 	fs: 'empty',
+			// },
```
Comment (Kporal) [inline tools/scripts-config-react-webpack/config/webpack.config.umd.js:123]:
```suggestion
```

File: tools/scripts-config-react-webpack/config/webpack.config.umd.js [hunk 5]
```diff
@@ -91,8 +95,16 @@ module.exports = options => {
...
+							// {
+							// 	loader: 'postcss-loader',
+							// 	options: {
+							// 		postcssOptions: {
+							// 			plugins: ['autoprefixer'],
+							// 		},
+							// 		sourceMap: true,
+							// 	},
+							// },
+							// { loader: 'resolve-url-loader', options: { sourceMap: true } },
```
Comment (Kporal) [inline tools/scripts-config-react-webpack/config/webpack.config.umd.js:107]: maybe we are missing from some css polyfill here?

Comment (jmfrancois) [reply tools/scripts-config-react-webpack/config/webpack.config.umd.js]: I have re-aligned with the app configuration

**Review by Kporal — APPROVED** [review]
👏 👏

**Pull request merged**